### PR TITLE
Add unit tests for DynamoDB integration

### PR DESCRIPTION
**Pull request by yito88**
https://scalar-labs.atlassian.net/browse/DLT-7304
To merge this branch into `integrate_dynamodb` (not master)

For the scan operation, we need to support multiple clustering keys, but I have not started it yet.
So, the current unit tests are temporary for the current implementation.

### CODE REVIEW

File: src/test/java/com/scalar/db/storage/dynamo/BatchHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        .doesNotThrowAnyException();
+
+    // Assert
+    ArgumentCaptor<TransactWriteItemsRequest> captor =
+        ArgumentCaptor.forClass(TransactWriteItemsRequest.class);
+    verify(client).transactWriteItems(captor.capture());
+    List<TransactWriteItem> items = captor.getValue().transactItems();
+    assertThat(items.size()).isEqualTo(4);
+    assertThat(items.get(0).put().item()).isEqualTo(dynamoMutation1.getValueMapWithKey());
+    assertThat(items.get(0).put().conditionExpression()).isEqualTo(null);
```
Comment (feeblefakie) [inline src/test/java/com/scalar/db/storage/dynamo/BatchHandlerTest.java:152]:
```suggestion
    assertThat(items.get(0).put().conditionExpression()).isNull();
```

File: src/test/java/com/scalar/db/storage/dynamo/BatchHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    assertThat(items.get(1).put().item()).isEqualTo(dynamoMutation2.getValueMapWithKey());
+    assertThat(items.get(1).put().conditionExpression())
+        .isEqualTo(dynamoMutation2.getIfNotExistsCondition());
+    assertThat(items.get(2).delete().key()).isEqualTo(dynamoMutation3.getKeyMap());
+    assertThat(items.get(2).delete().conditionExpression()).isEqualTo(null);
```
Comment (feeblefakie) [inline src/test/java/com/scalar/db/storage/dynamo/BatchHandlerTest.java:157]:
```suggestion
    assertThat(items.get(2).delete().conditionExpression()).isNull();
```

**Review by feeblefakie — APPROVED** [review]
LGTM! 
Just left very minor suggestions with isNull.

File: src/test/java/com/scalar/db/storage/dynamo/DeleteStatementHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+              handler.handle(delete);
+            })
+        .doesNotThrowAnyException();
+
+    // Assert
+    ArgumentCaptor<DeleteItemRequest> captor = ArgumentCaptor.forClass(DeleteItemRequest.class);
+    verify(client).deleteItem(captor.capture());
+    DeleteItemRequest actualRequest = captor.getValue();
+    assertThat(actualRequest.key()).isEqualTo(dynamoMutation.getKeyMap());
+    assertThat(actualRequest.conditionExpression()).isEqualTo(null);
```
[the remark this answers is not in the document]
Comment (yito88) [reply src/test/java/com/scalar/db/storage/dynamo/DeleteStatementHandlerTest.java]: Thank you! I fixed it!

**Pull request merged**